### PR TITLE
Added AllSRV and MaybeSRV

### PR DESCRIPTION
**Pull request by jameshartig**

### CODE REVIEW

File: srvclient.go [hunk 1]
```diff
@@ -38,13 +53,14 @@ func lookupSRV(hostname string) (*dns.Msg, error) {
 	m.SetQuestion(fqdn, dns.TypeSRV)
 	m.SetEdns0(dns.DefaultMsgSize, false)
 
+	var res *dns.Msg
 	for _, server := range cfg.servers {
-		res, _, err := c.Exchange(m, server+":53")
+		res, _, err = c.Exchange(m, server+":53")
```
Comment (mediocregopher) [inline srvclient.go]: :sheep: you can combine this into the `if` below it

File: srvclient.go [hunk 2]
```diff
@@ -53,11 +69,24 @@ func lookupSRV(hostname string) (*dns.Msg, error) {
...
+	if len(res.Answer) == 0 {
+		return nil, fmt.Errorf("No SRV records for %q", hostname)
+	}
+
+	ans := make([]*dns.SRV, len(res.Answer))
+	for i := range res.Answer {
+		if ansSRV, ok := res.Answer[i].(*dns.SRV); ok {
+			ans[i] = ansSRV
+		}
+	}
```
Comment (mediocregopher) [inline srvclient.go:82]: should you add a check at this point to make sure `len(ans) > 0`? If so you probably don't even need the one above this for loop

File: srvclient.go [hunk 3]
```diff
@@ -113,6 +131,44 @@ func SRVNoPort(hostname string) (string, error) {
 	return addr[:strings.Index(addr, ":")], nil
 }
 
+// AllSRV returns the list of all hostnames
```
Comment (mediocregopher) [inline srvclient.go]: We should specify that they will be sorted by weight/priority in the comment

File: srvclient.go (hunk 2 above)
Comment (jameshartig) [inline srvclient.go:82]: sounds good

File: srvclient.go (hunk 3 above)
Comment (jameshartig) [inline srvclient.go]: oh woops ya i didn't finish writing this comment >_<

File: srvclient.go [hunk 4]
```diff
@@ -113,6 +131,44 @@ func SRVNoPort(hostname string) (string, error) {
...
+		if ogPort != "" {
+			res[i] = ans[i].Target + ":" + ogPort
+		} else {
+			res[i] = ans[i].Target + ":" + strconv.Itoa(int(ans[i].Port))
+		}
+	}
+	return res, nil
+}
+
+// MaybeSRV attemps a SRV lookup if the host doesn't contain a port and if the
```
Comment (mediocregopher) [inline srvclient.go]: attempts :sheep:

File: srvclient.go [hunk 5]
```diff
@@ -113,6 +128,45 @@ func SRVNoPort(hostname string) (string, error) {
 	return addr[:strings.Index(addr, ":")], nil
 }
 
+// AllSRV returns the list of all hostnames and ports for the SRV lookup
+// The results are sorted by priority and then weight.
```
Comment (mediocregopher) [inline srvclient.go]: Sorry, one last thing, we should document that this has the same port behavior as the `SRV` method, but it replaces the port on all results.

**Pull request merged**